### PR TITLE
[ADD] Updation of the inherited fields information in the Top-down approach

### PR DESCRIPTION
**Pull request by jananjoy**

### CODE REVIEW

File: project_event/models/project_project.py [hunk 1]
```diff
@@ -55,6 +55,16 @@ def _compute_event_log_count(self):
                 ('res_id', '=', rec.id)
             ])
 
+    @api.multi
+    def write(self, vals):
+        activity = self.env['project.task'].search([
```
Comment (rimbendhaou) [inline project_event/models/project_project.py]: @jananjoy self.task_ids returns all the activities

File: project_event/models/project_project.py [hunk 2]
```diff
@@ -55,6 +55,16 @@ def _compute_event_log_count(self):
...
+            ('activity_task_type', '=', 'activity'),
+            ('project_id', '=', self.name)
+        ])
+        for rec in activity:
+            rec.write(vals)
+        return super(Project, self).write(vals)
```
Comment (rimbendhaou) [inline project_event/models/project_project.py]: better to start by super(Project, self).write(vals)

File: project_event/models/project_project.py [hunk 3]
```diff
@@ -55,6 +55,16 @@ def _compute_event_log_count(self):
                 ('res_id', '=', rec.id)
             ])
 
+    @api.multi
+    def write(self, vals):
```
Comment (rimbendhaou) [inline project_event/models/project_project.py:59]: Why you override this method?

File: project_event/models/project_project.py [hunk 4]
```diff
@@ -55,6 +55,16 @@ def _compute_event_log_count(self):
...
+        activity = self.env['project.task'].search([
+            ('activity_task_type', '=', 'activity'),
+            ('project_id', '=', self.name)
+        ])
+        for rec in activity:
+            rec.write(vals)
```
Comment (rimbendhaou) [inline project_event/models/project_project.py]: not always evident that we will have the same field name for project and task

File: project_event/models/project_task.py [hunk 5]
```diff
@@ -140,6 +140,18 @@ def _compute_asterisk_column(self):
             else:
                 rec.asterisk_validate_record = " "
 
+    @api.onchange('project_id')
+    def onchange_project(self):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: onchange_project_id

File: project_event/models/project_task.py [hunk 6]
```diff
@@ -140,6 +140,18 @@ def _compute_asterisk_column(self):
...
+        activity = self.env['project.project'].browse(self.project_id.id)
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: activity or project?

File: project_event/models/project_task.py [hunk 7]
```diff
@@ -140,6 +140,18 @@ def _compute_asterisk_column(self):
...
+        self.responsible_id = activity.responsible_id
+        self.partner_id = activity.partner_id
+
+    @api.onchange('parent_id')
+    def onchange_parent(self):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: def onchange_parent_id

File: project_event/models/project_task.py [hunk 8]
```diff
@@ -140,6 +140,18 @@ def _compute_asterisk_column(self):
...
+        task = self.env['project.task'].browse(self.parent_id.id)
+        self.responsible_id = task.responsible_id
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: self.responsible_id = self.parent_id.responsible_id

File: project_event/models/project_task.py [hunk 9]
```diff
@@ -140,6 +140,18 @@ def _compute_asterisk_column(self):
...
 
+    @api.onchange('project_id')
+    def onchange_project(self):
+        activity = self.env['project.project'].browse(self.project_id.id)
+        self.responsible_id = activity.responsible_id
+        self.partner_id = activity.partner_id
+
+    @api.onchange('parent_id')
+    def onchange_parent(self):
+        task = self.env['project.task'].browse(self.parent_id.id)
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: parent task is self.parent_id

File: project_event/models/project_task.py [hunk 10]
```diff
@@ -140,6 +140,18 @@ def _compute_asterisk_column(self):
...
+        self.responsible_id = task.responsible_id
+        self.partner_id = task.partner_id
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: self.partner_id = self.parent_id.partner_id

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_project.py [hunk 11]
```diff
@@ -55,6 +55,17 @@ def _compute_event_log_count(self):
...
             ])
 
+    @api.multi
+    def write(self, vals):
+        super(Project, self).write(vals)
+        activities = self.env['project.task'].search([
+            ('activity_task_type', '=', 'activity'),
+            ('project_id', '=', self.name)
+        ])
+        for rec in activities:
```
Comment (rimbendhaou) [inline project_event/models/project_project.py]: for rec in self.task_ids:

File: project_event/models/project_project.py [hunk 12]
```diff
@@ -55,6 +55,17 @@ def _compute_event_log_count(self):
                 ('res_id', '=', rec.id)
             ])
 
+    @api.multi
+    def write(self, vals):
+        super(Project, self).write(vals)
+        activities = self.env['project.task'].search([
```
Comment (rimbendhaou) [inline project_event/models/project_project.py]: no need to uese the research method

File: project_event/models/project_task.py [hunk 13]
```diff
@@ -140,6 +140,17 @@ def _compute_asterisk_column(self):
             else:
                 rec.asterisk_validate_record = " "
 
+    @api.onchange('project_id')
+    def onchange_project_id(self):
+        self._onchange_project()
+        self.responsible_id = self.project_id.responsible_id
+        self.partner_id = self.project_id.partner_id
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: it already do this in the parent _onchange_project

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_project.py [hunk 14]
```diff
@@ -55,6 +55,13 @@ def _compute_event_log_count(self):
                 ('res_id', '=', rec.id)
             ])
 
+    @api.multi
+    def write(self, vals):
+        super(Project, self).write(vals)
+        for rec in self.task_ids:
+            rec.write({'responsible_id': str(self.responsible_id.id),
```
Comment (rimbendhaou) [inline project_event/models/project_project.py]: Why str()?

Comment (rimbendhaou) [reply project_event/models/project_project.py]: What if self.responsible_id (or parent_id) is empty?

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_project.py (hunk 14 above)
Comment (jananjoy) [reply project_event/models/project_project.py]: @rimbendhaou: Done

**Review by eilst — APPROVED** [review]

File: project_event/models/project_project.py [hunk 15]
```diff
@@ -55,6 +55,13 @@ def _compute_event_log_count(self):
                 ('res_id', '=', rec.id)
             ])
 
+    @api.multi
+    def write(self, vals):
```
Comment (rimbendhaou) [inline project_event/models/project_project.py:59]: It doesnt update the responsible and partner of tasks of the event activities

**Review by rimbendhaou — DISMISSED** [review]

File: project_event/models/project_project.py (hunk 2 above)
Comment (jananjoy) [reply project_event/models/project_project.py]: @rimbendhaou  Done

File: project_event/models/project_project.py (hunk 4 above)
Comment (jananjoy) [reply project_event/models/project_project.py]: @rimbendhaou : Change the code accordingly

File: project_event/models/project_task.py (hunk 5 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou Done

File: project_event/models/project_task.py (hunk 6 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou  Change it with task_ids

File: project_event/models/project_task.py (hunk 7 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou Done

File: project_event/models/project_task.py (hunk 8 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou  Done

File: project_event/models/project_task.py (hunk 9 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou  Done

File: project_event/models/project_task.py (hunk 10 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou Done

File: project_event/models/project_project.py (hunk 11 above)
Comment (jananjoy) [reply project_event/models/project_project.py]: @rimbendhaou  Change accordingly

File: project_event/models/project_project.py (hunk 12 above)
Comment (jananjoy) [reply project_event/models/project_project.py]: @rimbendhaou :+1:

File: project_event/models/project_task.py (hunk 13 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou  Ok

File: project_event/models/project_project.py (hunk 15 above)
Comment (jananjoy) [reply project_event/models/project_project.py:59]: @rimbendhaou Done

File: project_event/models/project_project.py (hunk 2 above)
Comment (eilst) [reply project_event/models/project_project.py]: @rimbendhaou Not necessary in python3

**Review by eilst — APPROVED** [review]

**Pull request merged**